src/hooks/useTimeRemainingInBatch.ts

```diff
@@ -1,13 +1,19 @@
 import useSafeState from './useSafeState'
-import { getSecondsRemainingInBatch } from 'utils'
+import { getTimeRemainingInBatch } from 'utils'
 import { useEffect, useRef } from 'react'
 import { BATCH_TIME } from 'const'
 
 export function useTimeRemainingInBatch(): number {
-  const [timeRemaining, setTimeRemaining] = useSafeState(getSecondsRemainingInBatch())
+  const [timeRemaining, setTimeRemaining] = useSafeState(getTimeRemainingInBatch())
 
   useEffect(() => {
-    const interval = setInterval(() => setTimeRemaining(getSecondsRemainingInBatch()), 1000)
+    // timeout to start the timer exactly at half second
+    let interval = setTimeout(() => {
+      // update once
+      setTimeRemaining(getTimeRemainingInBatch())
+      // update every second from now on
+      interval = setInterval(() => setTimeRemaining(getTimeRemainingInBatch()), 1000)
+    }, Date.now() % 500)
```

Comment:
if `Date.now() % 500 === 0` better not schedule an update but update right away. Otherwise there's a delay because of task queuing. 
Check with
```js
function ttime() {
    const now = Date.now()
    setTimeout(() => console.log('delay', Date.now() - now), 0)
}
```
but know that our delay will be longer because of all sync code on stack that needs to run first, then the microtask queue that needs to empty, then any older tasks from the tasks queue

Reply:
> if Date.now() % 500 === 0 better not schedule an update but update right away. 

Hmm yes makes sense.

> but know that our delay will be longer because of all sync code on stack that needs to run first

About this, any better way of syncing all of these countdowns?

Reply:
requestAnimationFrame would be more precise, but also more annoying
setTimeout is good enough